mirrord/layer/src/go/linux_aarch64.rs

```diff
@@ -0,0 +1,303 @@
+use std::arch::asm;
+
+use errno::errno;
+use tracing::trace;
+
+use crate::{
+    close_detour, file::hooks::*, hooks::HookManager, macros::hook_symbol, socket::hooks::*,
+    FILE_MODE,
+};
+
+/// [Naked function] maps to gasave_systemstack_switch, called by asmcgocall.abi0
+#[no_mangle]
+#[naked]
+unsafe extern "C" fn gosave_systemstack_switch() {
+    asm!(
+        "adrp       x0,0x73000",
+        "add        x0,x0,0x3d0",
+        "add        x0,x0,0x8",
+        "str        x0,[x28, 0x40]",
+        "mov        x0,sp",
+        "str        x0,[x28, 0x38]",
+        "str        x29,[x28, 0x68]",
+        "str        xzr,[x28, 0x60]",
+        "str        xzr,[x28, 0x58]",
+        "ldr        x0,[x28, 0x50]",
+        "cbz        x0,1f",
+        "bl         go_runtime_abort",
+        "1:",
+        "ret",
+        options(noreturn)
+    );
+}
+
+/// Syscall & Syscall6 handler - supports upto 6 params, mainly used for
+/// accept4 Note: Depending on success/failure Syscall may or may not call this handler
+#[no_mangle]
+unsafe extern "C" fn c_abi_syscall6_handler(
```

Comment:
If there's a way to share this code between the architectures instead of duplicating it that would be nice.

Reply:
ah right, I wanted to do it but forgot.